compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -379,21 +379,21 @@ fn index_crate<'a>(
         fn visit_item(&mut self, item: &'a ast::Item) {
             let def_id = self.node_id_to_def_id[&item.id];
             *self.index.ensure_contains_elem(def_id, || AstOwner::NonOwner) = AstOwner::Item(item);
-            visit::walk_item(self, item)
+            visit::walk_item(self, item, ())
```

Comment:
The `()` stuff is an implementation detail that shouldn't generally leak outside of the (mut) visit modules, IMO.
There should be a wrapper hiding it, and `walk_item` was such a wrapper (probably still can be).

Reply:
I changed the general (with ctxt) function from `walk_item` to `walk_item_ctxt` and added `walk_item` and `walk_assoc_item` wrappers